File: python/hamming/hamming.py
def distance(strand_a: str, strand_b: str) -> int:
    if len(strand_a) != len(strand_b):
        raise ValueError("two strands length mismatch")
        
    char: int = 0
    for i in range(len(strand_a)):
        char = char if strand_a[i] == strand_b[i] else char + 1
    return char

File: go/hamming/hamming.go
package hamming

import "errors"

// Distance calculates the hamming distance of two ascii strings.
func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return 0, errors.New("length of sequences mismatch")
	}
	var c int
	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			c++
		}
	}
	return c, nil
}
